### PR TITLE
added detailed code for the function plot_sampling_hist

### PR DESCRIPTION
**Pull request by YueJiangMDSV**
Hi Team,

In this pull request, I added detailed code for the function plot_sampling_hist and added tests to handle incorrect function inputs. Please review it. Thanks!

Alex

### CODE REVIEW

File: R/plot_sampling_hist.R [hunk 1]
```diff
@@ -17,5 +17,53 @@
...
 #' plot_sampling_hist(pop, samples, var_name, c(1, 10), 3)
 plot_sampling_hist <- function(pop, samples, var_name, n_s, reps){
+  if (!"data.frame" %in% c(class(samples))) stop("Samples should be a data frame or a tibble")
+  if (!is.numeric(reps) == TRUE) stop("Number of replications should be a numerical value, a vector with length 1")
+  for (i in n_s){
+    if (class(i) != "numeric") stop("Samples' sizes should be a list or a vector with only numeric values")
+  }
+
+  samples <-
+    samples %>%
```
Comment (tguo9) [inline R/plot_sampling_hist.R:44]: You should add %>% operator to the global function definition

Comment (tguo9) [reply R/plot_sampling_hist.R:44]: Same for all %>%

File: R/plot_sampling_hist.R [hunk 2]
```diff
@@ -17,5 +17,53 @@
...
+    ungroup()
+
+  x_min <-
+    samples %>%
+    select({{var_name}}) %>%
+    unlist() %>%
+    quantile(0.05)
```
Comment (tguo9) [inline R/plot_sampling_hist.R]: The `ungroup`, `select` and `quantile`need to use :: notation

File: R/plot_sampling_hist.R [hunk 3]
```diff
@@ -17,5 +17,53 @@
...
+    group_by(replicate, size, rep_size) %>%
+    summarise(mean = mean({{var_name}}))
+
+
+  sampling_dist<- list()
+  for (i in 1:length(n_s)){
+    sampling_dist[[ i ]] <-
+      summary %>%
+      filter(size == n_s[i]) %>%
+      ggplot() + geom_histogram(aes(mean, ..density..)) +
```
Comment (tguo9) [inline R/plot_sampling_hist.R]: I think you need to add ggplot to the dependence list
refer: https://r-pkgs.org/whole-game.html#use_package

**Review by tguo9 — CHANGES_REQUESTED** [review]
Please check the comments. Thanks!

File: R/plot_sampling_hist.R (hunk 1 above)
Comment (YueJiangMDSV) [reply R/plot_sampling_hist.R:44]: Thanks for pointing out!

File: R/plot_sampling_hist.R (hunk 2 above)
Comment (YueJiangMDSV) [reply R/plot_sampling_hist.R]: Good catch! Thanks! I'll update those.

File: R/plot_sampling_hist.R (hunk 3 above)
Comment (YueJiangMDSV) [reply R/plot_sampling_hist.R]: Thanks! I'll take a look at it.

File: R/plot_sampling_hist.R [hunk 4]
```diff
@@ -17,5 +17,53 @@
...
+      theme(plot.title = element_text(size = 10))
+    if (i > 1){
+      sampling_dist[[i]] <- sampling_dist[[i]] + theme(axis.title.y=element_blank(),
+                                                       axis.text.y=element_blank(),
+                                                       axis.ticks.y=element_blank())
+    }
+
+  }
+
+  return(grid.arrange(grobs = sampling_dist, nrow=1, top = "Sampling Distribution Histograms"))
```
Comment (hwilliams10) [inline R/plot_sampling_hist.R]: Note:  it is difficult to do tests on grid.arrange / grob objects. I ended up splitting my code into two functions: one that creates the histograms (so you can test labels, axis mapping, etc), and one that arranges them in a grid

Comment (hwilliams10) [reply R/plot_sampling_hist.R]: Not required, just a suggestion

File: R/plot_sampling_hist.R (hunk 3 above)
Comment (tguo9) [reply R/plot_sampling_hist.R]: Looks good

File: R/plot_sampling_hist.R (hunk 2 above)
Comment (tguo9) [reply R/plot_sampling_hist.R]: Looks good

File: R/plot_sampling_hist.R (hunk 4 above)
Comment (YueJiangMDSV) [reply R/plot_sampling_hist.R]: updated it! Thanks!

**Review by tguo9 — COMMENTED** [review]
Everything looks good to me! @hwilliams10, do you have any comments for the testing part?

**Comment by hwilliams10** [comment]
> Everything looks good to me! @hwilliams10, do you have any comments for the testing part?

I can't see anything that stands out. @YueJiangMDSV have you tried running `check()` and made sure that all tests pass?

**Comment by YueJiangMDSV** [comment]
Hi @hwilliams10 , I just updated the scripts again. Passed `check()` in Rstudio

**Comment by hwilliams10** [comment]
> Hi @hwilliams10 , I just updated the scripts again. Passed `check()` in Rstudio

Awesome! @tguo9 I think you have to confirm that your requested changes were completed before we can pull

**Review by tguo9 — APPROVED** [review]
Looks good!

**Pull request merged**